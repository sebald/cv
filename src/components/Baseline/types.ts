export interface BaselineGridState {
  show: boolean;
}
