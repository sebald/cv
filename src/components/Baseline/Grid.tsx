import * as React from 'react';
import { createPortal } from 'react-dom';

import { Baseline } from './Baseline';
import { BaselineGridState } from './types';

export class BaselineGrid extends React.Component<{}, BaselineGridState> {
  el: HTMLElement;
  state: BaselineGridState = {
    show: false,
  };

  constructor(props: object) {
    super(props);
    this.el = document.createElement('div');
  }

  handleKeyPress = (ev: KeyboardEvent) => {
    const modifier = ev.metaKey || ev.ctrlKey;
    if (modifier && ev.key === 'b') {
      this.setState(prevState => ({ show: !prevState.show }));
    }
  };

  componentDidMount() {
    document.body.appendChild(this.el);
    window.addEventListener('keypress', this.handleKeyPress);
  }

  componentWillUnmount() {
    document.body.removeChild(this.el);
    window.removeEventListener('keypress', this.handleKeyPress);
  }

  render() {
    return this.state.show ? createPortal(<Baseline />, this.el) : null;
  }
}
