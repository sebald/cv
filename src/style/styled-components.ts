import * as styledComponents from 'styled-components';
import {
  ThemedStyledComponentsModule,
  ThemedStyledFunction,
} from 'styled-components';

import { Theme } from './types';

// https://github.com/styled-components/styled-components/issues/630#issuecomment-317172454
function withProps<U>() {
  return <P, T, O>(
    fn: ThemedStyledFunction<P, T, O>
  ): ThemedStyledFunction<P & U, T, O & U> =>
    fn as ThemedStyledFunction<P & U, T, O & U>;
}

// Alternative, maybe better approach?
type TTag = keyof JSX.IntrinsicElements;

function suit<U>(tag: TTag) {
  return styled[tag] as ThemedStyledFunction<
    JSX.IntrinsicElements[TTag] & U,
    Theme
  >;
}

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider,
} = styledComponents as ThemedStyledComponentsModule<Theme>;

export { styled, css, injectGlobal, keyframes, withProps, suit, ThemeProvider };
