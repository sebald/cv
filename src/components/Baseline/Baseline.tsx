import { baseline, styled } from '../../style';

export const Baseline = styled.div`
  position: absolute;
  top: -1px;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 9999;

  background-image: repeating-linear-gradient(
    to bottom,
    rgba(255, 105, 180, 0.5),
    rgba(255, 105, 180, 0.5) 1px,
    transparent 1px,
    transparent ${baseline(1)}
  );
`;
Baseline.displayName = 'Baseline';
