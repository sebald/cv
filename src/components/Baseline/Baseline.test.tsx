import { mount } from 'enzyme';
import * as React from 'react';
import { ThemeProvider } from 'styled-components';

import { Baseline } from './Baseline';

test('render default', () => {
  const wrapper = mount(
    <ThemeProvider theme={{ baseline: 24 }}>
      <Baseline />
    </ThemeProvider>
  );
  expect(wrapper).toMatchSnapshot();
});
