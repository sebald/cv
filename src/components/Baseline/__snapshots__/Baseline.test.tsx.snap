// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render default 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
  background-image: repeating-linear-gradient( to bottom, rgba(255,105,180,0.5), rgba(255,105,180,0.5) 1px, transparent 1px, transparent 1.5rem );
}

<ThemeProvider
  theme={
    Object {
      "baseline": 24,
    }
  }
>
  <Baseline>
    <div
      className="c0"
    />
  </Baseline>
</ThemeProvider>
`;
